packages/scripts/src/migration/dvm2-designated-voting/1_Migrate.ts

```diff
@@ -0,0 +1,168 @@
+import hre from "hardhat";
+import { utils } from "ethers";
+import { VotingTokenEthers, DesignatedVotingV2FactoryEthers, DesignatedVotingV2Ethers } from "@uma/contracts-node";
+import { getContractInstance } from "../../utils/contracts";
+
+import {
+  getDesignatedVotingContractsOwnedByOwner,
+  baseSafePayload,
+  appendTxToSafePayload,
+  GnosisTransaction,
+} from "./common";
+
+import fs from "fs";
+import path from "path";
+
+async function main() {
+  console.log("Running Migration Gnosis payload builder 👷‍♀️");
+  const chainId = Number(await hre.getChainId());
+  if (chainId != 1) throw new Error("Can only run on mainnet");
+
+  if (!process.env.OWNER_TO_MIGRATE) throw new Error("No OWNER_TO_MIGRATE set");
+  const owner = process.env.OWNER_TO_MIGRATE || "";
+
+  if (!process.env.GNOSIS_SAFE) throw new Error("No GNOSIS_SAFE set");
+  const safe = process.env.GNOSIS_SAFE || "";
+
+  // Step 1: fetch all current DesignatedVoting contracts owned by the owner. Remove elements that have 0 balance.
+  const designatedVotingData = (await getDesignatedVotingContractsOwnedByOwner(owner)).filter((e) => e.balance.gt(0));
+
+  // Step 2: fetch all the deployed DesignatedVotingV2 contracts and append them to this datastructure.
```

Comment:
the comment seems bit off as the code below only gets DesignatedVotingV2Factory contract, not all the deployed DesignatedVotingV2 contracts